.github/workflows/ci.yml

```diff
@@ -145,15 +145,9 @@ jobs:
       matrix:
         os:
           - ubuntu-latest
```

Comment:
NIT

The ubuntu-only job could be cleaned up from all `matrix.os` and `matrix.allow-failure` uses.

Reply:
After some thinking I'd like to keep it here for clarity in the job overview etc. Then all unitests are visible in the same way etc.

Reply:
I think we don't need a matrix definition here at all.

The job now only runs on `ubuntu-latest` with `allow-failure` always being true; not using a matrix would allow to simplify the job definition.